workbench-core/datasets/src/dataSetService.ts

```diff
@@ -50,16 +52,30 @@ export class DataSetService {
    * @returns the DataSet object which is stored in the backing datastore.
    */
   public async provisionDataSet(request: CreateProvisionDatasetRequest): Promise<DataSet> {
-    const { storageProvider, ...dataSet } = request;
-
-    await storageProvider.createStorage(dataSet.storageName, dataSet.path);
-
-    const provisioned: DataSet = {
-      ...dataSet,
-      storageType: storageProvider.getStorageType()
+    const metadata: Metadata = {
+      action: this.provisionDataSet.name,
+      source: {
+        serviceName: DataSetService.name
+      },
+      requestBody: request
     };
 
-    return await this._dbProvider.addDataSet(provisioned);
+    try {
+      const { storageProvider, ...dataSet } = request;
+
+      await storageProvider.createStorage(dataSet.storageName, dataSet.path);
+
+      const provisioned: DataSet = {
+        ...dataSet,
+        storageType: storageProvider.getStorageType()
+      };
+      const response: DataSet = await this._dbProvider.addDataSet(provisioned);
+      await this._audit.write(metadata, response);
```

Comment:
metadata needs to be updated with statusCode

Reply:
If it _needs_ to be here, why is it optional on the metadata object?
I don't know what status code will be returned yet. I am not doing a res.send() here.
200/400 are not necessarily accurate and will make the logs confusing.
what do?

Reply:
So it is optional as an input, but the auditService will verify if the audit entry is complete before writing it. This is to ensure all audits are following best practices and creating a valid audit trail. By default, auditService will require actor, source, statusCode, and action

Reply:
so you're saying that I will need to make a breaking change on every public API to add auditing?

Reply:
not necessarily, actor is dependent on what you determine is sufficient enough to do an audit trial on. If we require the user that is requesting this information to be the actor then yes, we will need that information.